posthog/decorators.py

```diff
@@ -0,0 +1,69 @@
+import json
+import hashlib
+from posthog.models import Filter
+from django.core.cache import cache
+import json
+
+def generate_cache_key(obj):
+    stringified = json.dumps(obj)
+    return hashlib.md5(stringified.encode("utf-8")).hexdigest()
+
+
+TRENDS_ENDPOINT = 'Trends'
+FUNNEL_ENDPOINT = 'Funnel'
+
+def cached_function(cache_type: str, expiry=30):
```

Comment:
One thing to note is that if we have a default expiry of 30 seconds this will be nice for very heavily trafficked instances but for new instances, it might seem like the data is not refreshing/stuck when you're on the trends page. I'm open to limiting the caching for dashboart items only